Data Table: Bot Accounts

ChrisHuynh333 opened the pull request:
## What does this PR do and why?
Note: Due to the size of this current PR, the following will be branched out and fixed in separate PRs:
- Removal of new PAT panel upon bot deletion and PAT revoke

This PR refactors the bot account pages (Group and Project) to use the `data_table_component` (includes PAT tables to use `data_table_component`). This required much of the response logic (turbo etc.) to be re-done. 

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/5902493a-e704-4ea1-a189-e19e1d279cf4)

![image](https://github.com/user-attachments/assets/793cd714-2826-42ed-98e2-0b53b2bdd98f)

## How to set up and validate locally
Test all bot functionality:
1. Create new bot (table updates with new bot and new PAT panel appears/updates)
2. Delete bot (table updates, empty state appears if there are 0 bots)
3. Active PATs for a bot and Revoke action (empty state of PAT table once all PATs are revoked)
4. Generate new PAT (table active PAT # updates, new PAT panel appears/updates)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

joshsadam reviewed (changes requested):
This looks really great overall.  I actually only have one wording change I would like to see.

ChrisHuynh333 commented on config/locales/en.yml:
```diff
@@ -311,6 +311,34 @@ en:
+        actions:
+          generate_new_token: Generate new token
+          generate_new_token_aria_label: Generate a new personal access token for bot account
+          remove: Delete bot
```
Agreed, changed in 90e8c7ed61effaa857e6a5de0cb6f0ef6097b8e2.

I've also changed the key from `remove` to `destroy` as that lines up more with what we've done for other translations.

deepsidhu85 reviewed (dismissed):
Looks good to me!

joshsadam commented:
I found a minor UI bug that I am hoping you can fix, and I am hoping that I can explain it properly. I will paste the screenshot and try to explain below.
![bots](https://github.com/user-attachments/assets/fbce4564-4850-447b-a44f-8c8f372696aa)

- The div highlighted in the developer console wraps a `turbo-frame`.
- When the turbo frame is empty (as in the screenshot) the div is empty but still renders and has the class of `mb-4` which for all purposes makes the margin bottom on the header 8.
- Can you remove the wrapping div and set the `mb-4` on the content within the `turbo-frame` so that it is only there if the `turbo-frame` has content?

joshsadam commented:
I think this small bug is something that you can quickly fix in this PR.  The create  bot modal body does not vertically scroll.  If you don't have enough vertical screen space you cannot fill in all the values.  As yo can see in the screenshot I cannot select a date 
![image](https://github.com/user-attachments/assets/5f5fc71c-a55d-45d7-84a9-005cda56535f)

joshsadam reviewed (changes requested):
This looks good.  I left 2 comments that I am hoping you can quickly fix. Good work on this.

ChrisHuynh333 commented:
> I found a minor UI bug that I am hoping you can fix, and I am hoping that I can explain it properly. I will paste the screenshot and try to explain below. ![bots](https://private-user-images.githubusercontent.com/11295750/397177725-fbce4564-4850-447b-a44f-8c8f372696aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MjE0ODQsIm5iZiI6MTczNDYyMTE4NCwicGF0aCI6Ii8xMTI5NTc1MC8zOTcxNzc3MjUtZmJjZTQ1NjQtNDg1MC00NDdiLWE0NGYtOGM4ZjM3MjY5NmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDE1MTMwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMjAxMDZkM2EwYmRmOTcwMzI1MWFkNTg1YzQwNmE5NGE3NDA0ZjU4M2RkNTZiYjI0NjI2OTVhZTRjYTcyNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4jXUXfZTLdsDrabfFuXL5fSAbZD1m8p-vRDHM9aKXsQ)
> 
> * The div highlighted in the developer console wraps a `turbo-frame`.
> * When the turbo frame is empty (as in the screenshot) the div is empty but still renders and has the class of `mb-4` which for all purposes makes the margin bottom on the header 8.
> * Can you remove the wrapping div and set the `mb-4` on the content within the `turbo-frame` so that it is only there if the `turbo-frame` has content?

Thanks, removed the wrapper and moved the `mb-4` to the frame content for both projects and groups in f039c402c4192177e1202278e122a8123f25ffa6

joshsadam reviewed (approved):
This looks good to me
